src/platform/log-web.d.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * @fileoverview A TS declaration for log api.
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+export declare function logFactory(...args:any):(...args:any) => void;
```

Comment:
That's quite liberal :D

Reply:
I got lazy, the actual return type is function(preamble, color, log = 'log'), but the logFactory expects it to be varargs, and the TS compiler kept choking trying to return a 4-arg function from a varargs function. 

But there's other d.ts files we're using that are similarly too liberal.